paddlenlp/experimental/transformers/opt/modeling.py

```diff
@@ -0,0 +1,516 @@
+            concated_qkv_weight = concated_qkv_weight.transpose(1, 0)
+            concated_qkv_weight = concated_qkv_weight.reshape(3 * self.num_heads * self.head_size, self.hidden_size)
+            concated_qkv_weight = paddle.to_tensor(concated_qkv_weight)
+
+            concated_qkv_bias = np.concatenate([q_bias, k_bias, v_bias], axis=-1)
+            concated_qkv_bias = concated_qkv_bias.reshape(3 * self.num_heads * self.head_size)
+            concated_qkv_bias = paddle.to_tensor(concated_qkv_bias)
+
+            out_proj_weight = paddle.to_tensor(state_dict["opt.decoder.layers.{}.self_attn.out_proj.weight".format(i)])
+            out_proj_bias =  paddle.to_tensor(state_dict["opt.decoder.layers.{}.self_attn.out_proj.bias".format(i)])
+
+            ffn_ln_scale = paddle.to_tensor(state_dict["opt.decoder.layers.{}.norm2.weight".format(i)])
+            ffn_ln_bias = paddle.to_tensor(state_dict["opt.decoder.layers.{}.norm2.bias".format(i)])
+            ffn_ln_scale = paddle.cast(ffn_ln_scale, "float32")
+            ffn_ln_bias = paddle.cast(ffn_ln_bias, "float32")
+
+            ffn1_weight = paddle.to_tensor(state_dict["opt.decoder.layers.{}.linear1.weight".format(i)])
+            ffn1_bias = paddle.to_tensor(state_dict["opt.decoder.layers.{}.linear1.bias".format(i)])
+            ffn2_weight = paddle.to_tensor(state_dict["opt.decoder.layers.{}.linear2.weight".format(i)])
+            ffn2_bias = paddle.to_tensor(state_dict["opt.decoder.layers.{}.linear2.bias".format(i)])
+
+            self.transformer_block.ln_scales[i].set_value(ln_scale)
+            self.transformer_block.ln_biases[i].set_value(ln_bias)
+
+            self.transformer_block.qkv_weights[i].set_value(concated_qkv_weight)
+            self.transformer_block.qkv_biases[i].set_value(concated_qkv_bias)
+
+            self.transformer_block.linear_weights[i].set_value(out_proj_weight)
+            self.transformer_block.linear_biases[i].set_value(out_proj_bias)
+
+            self.transformer_block.ffn_ln_scales[i].set_value(ffn_ln_scale)
+            self.transformer_block.ffn_ln_biases[i].set_value(ffn_ln_bias)
+
+            self.transformer_block.ffn1_weights[i].set_value(ffn1_weight)
+            self.transformer_block.ffn1_biases[i].set_value(ffn1_bias)
+
+            self.transformer_block.ffn2_weights[i].set_value(ffn2_weight)
+            self.transformer_block.ffn2_biases[i].set_value(ffn2_bias)
+
+
+class OPTForCausalLMInferenceModel(GenerationInferenceModel, OPTPretrainedModel):
+
+    def __init__(self, config: OPTConfig, **kwargs):
+        super(OPTForCausalLMInferenceModel, self).__init__(config)
+        self.opt = OPTInferenceModel(config)
+        self.lm_head = OPTLMHead(config)
+
+    @classmethod
+    def from_pretrained(
+        cls, pretrained_model_name_or_path, from_hf_hub: bool = False, subfolder: str | None = None, *args, **kwargs
+    ):
+        # TODO: Support safetensors loading.
+        kwargs["use_safetensors"] = False
```

Comment:
```suggestion
        kwargs["use_safetensors"] = kwargs.get("use_safetensors", False)
```

建议用这个代码，因为在单卡情况下，safetensor 还是可以用的。

Reply:
> kwargs["use_safetensors"] = kwargs.get("use_safetensors", False)

done